pkg/sql/jobs/registry_external_test.go

```diff
@@ -175,31 +174,35 @@ func TestRegistryResumeExpiredLease(t *testing.T) {
 		return nil
 	})
 
-	lock.Lock()
-	if e, a := 2, resumeCounts[jobMap[1]]; e != a {
-		t.Fatalf("expected resumeCount to be %d, but got %d", e, a)
-	}
-	lock.Unlock()
+	testutils.SucceedsSoon(t, func() error {
+		lock.Lock()
+		defer lock.Unlock()
+		if e, a := 2, resumeCounts[jobMap[1]]; e != a {
+			return errors.Errorf("expected resumeCount to be %d, but got %d", e, a)
+		}
+		return nil
+	})
 
 	nodeLiveness.FakeIncrementEpoch(3)
 	<-resumeCalled
+	close(done)
 	testutils.SucceedsSoon(t, func() error {
 		lock.Lock()
-		if e, a := 2, resumeCounts[jobMap[3]]; e != a {
+		defer lock.Unlock()
+		if e, a := 2, resumeCounts[jobMap[3]]; e > a {
 			return errors.Errorf("expected resumeCount to be %d, but got %d", e, a)
```

Comment:
Ideally the error message would be updated a la `expected hookCallCount to be > %d, but got %d`.